chain/chain/src/store.rs

```diff
@@ -2227,6 +2226,11 @@ impl<'a> ChainStoreUpdate<'a> {
         if hashes.is_empty() {
             epoch_to_hashes.remove(epoch_id);
         }
+        // TODO(mina86): Unfortunately, we need to convert this key to a vector
+        // because CellLruCache::pop expects `lru::KeyRef<K>: Borrow<Q>` (where
+        // `&Q` is the argument) and `K` is `Vec<u8>`.  I feel like there should
+        // be some way to be able to pass `&[u8]` (which would save us memory
+        // allocation) but for now keep `to_vec`.
```

Comment:
@matklad, do you have an idea how to deal with this? If I try to pass the key as a slice I get:

```
error[E0277]: the trait bound `KeyRef<Vec<u8>>: Borrow<[u8; 8]>` is not satisfied
    --> chain/chain/src/store.rs:2232:52
     |
2232 |             self.chain_store.block_hash_per_height.pop(&key);
     |                                                    ^^^ the trait `Borrow<[u8; 8]>` is not implemented for `KeyRef<Vec<u8>>`
     |
     = help: the following implementations were found:
               <KeyRef<Box<T>> as Borrow<T>>
               <KeyRef<K> as Borrow<K>>
               <KeyRef<String> as Borrow<str>>
               <KeyRef<Vec<T>> as Borrow<[T]>>
note: required by a bound in `CellLruCache::<K, V>::pop`
    --> /home/mpn/code/near/nearcore/utils/near-cache/src/cell.rs:68:25
     |
68   |         lru::KeyRef<K>: Borrow<Q>,
     |                         ^^^^^^^^^ required by this bound in `CellLruCache::<K, V>::pop`
```

Reply:
Would making it a proper slice, rather than a reference to an array help? Auto-deref is sadly not sufficiently advanced to see through a bunch of layers of generics :(. So something like:

```rust
let key = &index_to_bytes(height)[..];
self.chain_store.block_hash_per_height.pop(key); // <- note no reference here
```

Reply:
Yes, that solves it.